core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalFileReader.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal.ufs;
+
+import alluxio.exception.InvalidJournalEntryException;
+import alluxio.master.journal.JournalReader;
+import alluxio.master.journal.options.JournalReaderOptions;
+import alluxio.proto.journal.Journal;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.util.proto.ProtoUtils;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Implementation of {@link JournalReader} that reads journal entries from a journal file.
+ */
+@NotThreadSafe
+final class UfsJournalFileReader implements JournalReader {
```

Comment:
There is no relationship between UfsJournalReader and UfsJournalFileReader. I cannot think of a better name here... Do you have any suggestions in mind? It is a reader to read a journal file.